src/adaptation/precond.jl

```diff
@@ -2,50 +2,59 @@
 #### Robust online (co-)variance estimators.
 ####
 
+## Variance estimator
+
 abstract type VarEstimator{T} end
 
 # NOTE: this naive variance estimator is used only in testing
-mutable struct NaiveVar{T} <: VarEstimator{T}
+mutable struct NaiveVar{T<:AbstractVector{<:AbstractVecOrMat{<:AbstractFloat}}} <: VarEstimator{T}
```

Comment:
Maybe rewrite this to improve readability:

`NaiveVar{T<:AbstractVector{F}} where F<:AbstractVecOrMat{<:AbstractFloat}`